### PR TITLE
Historical Work Page | BE

### PR DESCRIPTION
**Pull request by engineervix**
[Link to Ticket](https://torchbox.monday.com/boards/1192293412/pulses/1370033202)

### Description of Changes Made

This refactors the existing Work Page model to pave way for the introduction of a new Work Page model based on the new designs.

**Key Changes**

- Rename `WorkPage` to `HistoricalWorkPage`
- Create `AbstractPageScreenShot` and `AbstractPageAuthor`, so that, upon subclassing, we can create a `page` `ParentalKey` to whichever `Page` model we choose. In this PR, we do this for the `HistoricalWorkPage` above. We'll be able to do this for the new `WorkPage`
- Remove some unused code (mainly tags on work pages)
- update references to `WorkPage` so that they now refer to `HistoricalWorkPage`
- Ensure that  `HistoricalWorkPage` cannot be created in Wagtail Admin (we want people to create the new `WorkPage`s instead)

### How to Test

In Wagtail Admin, go to Pages > Torchbox > Work: http://127.0.0.1:8000/admin/pages/6/. 

- You should see the old Work Pages there, and they should be called Historical Pages
- You should be able to edit the existing pages
- You should not be able to create a Historical Work Page

### Screenshots

<details>
  <summary>Expand to see more</summary>

image

</details>

### MR Checklist

- [x] Add a description of your pull request and instructions for the reviewer to verify your work.
- [x] If your pull request is for a specific ticket, link to it in the description.
- [x] Stay on point and keep it small so the merge request can be easily reviewed.
- [x] Tests and linting passes.

#### Unit tests

- [ ] Added
- [x] Not required

#### Documentation

- [ ] Updated build docs
- [ ] Updated editor guidelines (https://docs.google.com/document/d/1PAWccdQ4tfaZsrEWmpDhvP3GH5RRmBOARFVp4b-kje8/edit?usp=sharing)
- [x] Not required

#### Browser testing

- [ ] I have tested in the following browsers and environments (edit the list as required)
  - Latest version of Chrome on mac
  - Latest version of Firefox on mac
  - Latest version of Safari on mac
  - Safari on last two versions of iOS
  - Chrome on last two versions of Android
- [x] Not required

#### Data protection

- [ ] Not relevant
- [x] This adds new sources of PII and documents it and modifies Birdbath processors accordingly

#### Accessibility

- [ ] Automated WCAG 2.1 tests pass
- [ ] Manual WCAG 2.1 tests completed
- [ ] I have tested in a screen reader
- [ ] I have tested in high-contrast mode
- [ ] Any animations removed for prefers-reduced-motion
- [x] Not required

#### Sustainability

- [ ] Images are optimised and lazy-loading used where appropriate
- [ ] SVGs have been optimised
- [ ] Perfomance and transfer of data considered
- [ ] If JavaScript is needed alternatives have been considered
- [ ] New font variants have not been added
- [x] Not required

#### Pattern library

- [ ] The pattern library component for this template displays correctly, and does not break parent templates
- [ ] The styleguide is updated if relevant
- [x] Changes are not relevant the pattern library

### CODE REVIEW

File: tbx/work/models.py
Comment (ababic) [inline tbx/work/models.py:40]: Let's avoid creating duplicate versions of these models for each page type - it's unnecessary. Just ALWAYS use `wagtailcore.Page` in the `ParentalKey` - Which options appear in the Wagtail admin UI for each page type is controlled via panels.

**Comment by helenb** [comment]
@engineervix I have now merged the fe changes for the historical work page to the integration branch

File: tbx/work/models.py
Comment (helenb) [inline tbx/work/models.py:61]: I think we will need to rename the template file before this is merged the FE code has now been merged so if you rebase you can do this.

Comment (engineervix) [reply tbx/work/models.py:38]: Thanks @helenb I have rebased. Fromhttps://github.com/torchbox/torchbox.com/pull/49/files it seems the template is still `"patterns/pages/work/work_detail.html"` so I've not updated it here

File: tbx/work/models.py
Comment (engineervix) [reply tbx/work/models.py]: Thanks @ababic I've added a couple of commits to remove these extraneous models. 
I've also added a data migration to ensure that we don't lose any author info for existing pages.

File: tbx/work/models.py
Comment (helenb) [reply tbx/work/models.py:38]: I thought we ought to rename it for clarity and consistency with the model name

Comment (engineervix) [reply tbx/work/models.py:38]: Updated in 6e5c4ff

Comment (helenb) [reply tbx/work/models.py:38]: looks good!

**Review by helenb — APPROVED** [review]
I have tested and this is working well

File: tbx/work/models.py
Comment (helenb) [inline tbx/work/models.py:24]: is this screenshots something that is actually in use somewhere? I can't see the option to add them when editing the page

File: tbx/work/models.py
Comment (helenb) [inline tbx/work/models.py:41]: Suggestion: change the last sentence to "It is kept here to display the older work pages that were created prior to 2024."

File: tbx/work/models.py
Comment (engineervix) [reply tbx/work/models.py:24]: Yes it is, here's a screenshot:

<details><summary>Screenshot</summary>

image

</details>

File: tbx/work/models.py
Comment (engineervix) [reply tbx/work/models.py]: done in 98f6fe4

File: tbx/work/models.py
Comment (helenb) [reply tbx/work/models.py:24]: It's not displayed in the template anywhere, and wasn't in the previous site either.

Comment (engineervix) [reply tbx/work/models.py:24]: Oh ok, maybe it's something that was used a long time ago and was later changed

Comment (helenb) [reply tbx/work/models.py:24]: I wouldn't let this block merging - but we might need to check in with Lisa if this should be displayed, and whether it should be on the new work page.

**Pull request merged**
